src/directives/for.js

```diff
@@ -65,8 +65,9 @@ function parseForExpression(expression) {
     return res
 }
 
-function getIterationScopeVariables(iteratorNames, item, index, items) {
-    let scopeVariables = { [iteratorNames.item]: item }
+function getIterationScopeVariables(iteratorNames, item, index, items, scopeVariables) {
+    if (! scopeVariables) scopeVariables = {}
```

Comment:
```suggestion
function getIterationScopeVariables(iteratorNames, item, index, items, extraVars) {
    let scopeVariables = extraVars ? {...extraVars} : {}
```
It should fix the issue highlighted by @WL1981
Since the variable is an object, javascript passes the reference so on each iteration you are overriding the previous value. It doesn't happen at level 1 since the variable is null so you instantiate a brand new object each time.
I hope it makes sense (You need to recompile dist after applying the change).

Reply:
Great job! This change will fix [this](https://github.com/alpinejs/alpine/pull/357#issuecomment-614510539) issue. Also, it will pass all (122) tests.

Thank you @SimoTod

Reply:
I've been thinking of maybe better names for `extraVars` would `scopeInheritedVariables` be accurate (instead of `extraVars`)?

Reply:
> I've been thinking of maybe better names for `extraVars` would `scopeInheritedVariables` be accurate (instead of `extraVars`)?

By scope inherited you mean inherited from an outside scope I assume. That's not always the case though with every handler, for example the $dispatch function is just an extraVar, not necessarily a variable / piece of data from a higher level scope. I agree that it does need to be renamed though so clearly state *what* it is (even though `extraVars` does kind of do that).

Reply:
Thanks @WL1981 @SimoTod @HugoDF @ryangjchandler , good catch! Will add this fix.

Reply:
Yep, I'm not precious about the suggestion. I just wanted to highlight that object are passed by reference in js so the code needs to create a fresh copy everytime. I didn't want to use the same name to avoid confusion but you can just keep 1 variable and reassign it at line 1. Up to the author.

Reply:
Agree with all this, ignore the renaming if it's going to add knowledge overhead